torch/autograd/functional.py

```diff
@@ -690,11 +769,14 @@ def ensure_single_output_function(*inp):
         return out.squeeze()
 
     def jac_func(*inp):
+        if forward_ad:
+            # Ensure the graph is created (and not detached)
```

Comment:
What does this mean? Is that for the create_graph case?

Reply:
Currently for  in this wrapper make sure inputs have requires_grad=True, and still create_graph=True so that input's won't be detached in jacobian's preprocessing (which detaches when either create_graph=False or requires_grad of its input is False). This way avoids changing jacobian's API but has more overhead because we must create the graph.

Alternatively we could add a for_forward_ad_hessian flag which will skip the _grad_preprocess stage and avoid detaching the inputs while also avoiding the creation of the backward graph

Reply:
Ho this was done before the idea that detach applies both to forward and backward mode AD

Should we replace the `.detach()` by a 
```
with torch.no_grad():
    inp = inp.clone()
```
when forward AD is used?
I guess not and we should introduce a new API to detach backward grads only <- TODO for forward AD project